src/avrftdi.c

```diff
@@ -702,12 +702,17 @@ static int avrftdi_open(PROGRAMMER *pgm, const char *port) {
 
   write_flush(pdata);
 
-  if(pgm->baudrate) {
-    set_frequency(pdata, pgm->baudrate);
-  } else if(pgm->bitclock) {
-    set_frequency(pdata, (uint32_t) (1.0f/pgm->bitclock));
-  } else {
-    set_frequency(pdata, pgm->baudrate? pgm->baudrate: 150000);
+  if(pgm->extra_features & HAS_BITCLOCK_ADJ) {
+    if(pgm->baudrate && pgm->bitclock && (int) (1.0/pgm->bitclock) != pgm->baudrate)
+      pmsg_warning("both -b baud and -B bitrate set; ignoring -B\n");
+    set_frequency(pdata,
+      pgm->baudrate? pgm->baudrate:
+      pgm->bitclock? (int) (1.0/pgm->bitclock):
+      150000
```

Comment:
Wouldn't this cause an issue if avrdude.conf doesn't have HAS_BITCLOCK_ADJ for avrftdi programmers?
It looks like avrftdi relies on a bitclock value being set, regardless of whether -B is specified or not. This code won't set a bitclock if HAS_BITCLOCK_ADJ is missing, and I don't think this is the correct behaviour.

Reply:
Ahh, yes, this escaped my review.

Reply:
Resolved